qiskit/circuit/quantumcircuit.py

```diff
@@ -163,17 +163,15 @@ def __init__(self, *regs, name=None, global_phase=0, metadata=None):
 
             regs = tuple(int(reg) for reg in regs)  # cast to int
 
-        if name is None:
-            name = self.cls_prefix() + str(self.cls_instances())
-            if sys.platform != "win32" and not is_main_process():
-                name += '-{}'.format(mp.current_process().pid)
-        self._increment_instances()
-
-        if not isinstance(name, str):
+        self._base_name = self.cls_prefix() if name is None else name
+        if not isinstance(self._base_name, str):
             raise CircuitError("The circuit name should be a string "
                                "(or None to auto-generate a name).")
-
-        self.name = name
+        if name is None:
```

Comment:
Can `name` still be `None` after L166?

Reply:
That line sets doesn't change `name` but rather `self._base_name`, however I've tried simplifying the logic in https://github.com/Qiskit/qiskit-terra/pull/5851/commits/79956706a9be850b0cb857e3034db7cf31f1aa6d.